test/hasvalue.jl

```diff
@@ -0,0 +1,209 @@
+@testset "base getvalue + hasvalue" begin
+    @testset "basic NamedTuple" begin
+        nt = (a=[1], b=2, c=(x=3, y=[4], z=(; p=[(; q=5)])), d=[1.0 0.5; 0.5 1.0])
```

Comment:
```suggestion
        nt = (
            a=[1],
            b=2,
            c=(
                x=3,
                y=[4],
                z=(;p=[(; q=5)])
            ),
            d=[1.0 0.5; 0.5 1.0],
        )
```
This whole NamedTuple just made me laugh. I also found it really hard to read. I wrote that suggestion, but I guess autoformatter is just gonna brutalise it back into unreadability, so maybe this is futile.